diffrax/_integrate.py

```diff
@@ -782,7 +773,7 @@ def _promote(yi):
         terms = MultiTerm(*terms)
 
     # Error checking
-    if not _term_compatible(y0, args, terms, solver.term_structure):
+    if not _term_compatible(solver, y0, args, terms, solver.term_structure):
```

Comment:
Can we avoid passing both `solver` and `solver.term_structure` separately?

Reply:
So given that a few lines above that one needs to pass a term structure that does not come from `solver.term_structure`, I will keep two separate arguments: `term_structure` and `contr_kwargs`. But I will not pass the solver itself.